src/tools/r2rdump/SignatureType.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Reflection.Metadata;
+using System.Text;
+
+namespace R2RDump
+{
+    class SignatureType
+    {
+        /// <summary>
+        /// Indicates if the type is an array, reference, or generic
+        /// </summary>
+        public SignatureTypeFlags Flags { get; }
+
+        /// <summary>
+        /// Name of the object or primitive type
+        /// </summary>
+        public string TypeName { get; }
+
+        /// <summary>
+        /// The type of the instance for generic a type
+        /// </summary>
+        public R2RMethod.GenericElementTypes GenericInstance { get; set; }
```

Comment:
Is this property necessary? The only reference I see to it is when you set it to 0 in this file.

Reply:
GenericInstance saves the type it got parameterized to. It's set in the R2RMethod ctor and used when printing the type. The TypeName saves the placeholder generic type such as T. Would it be better to save the instance in TypeName, since we don't really use the placeholder?